src/server/protocol.ts

```diff
@@ -1063,6 +1063,12 @@ namespace ts.server.protocol {
          */
         canRename: boolean;
 
+        /**
+         * fileName to rename.
+         * If set, `getEditsForFileRename` should be called instead of `findRenameLocations`.
+         */
+        fileToRename?: string;

```

Comment:
this could be a folder as well.. e.g. 

```ts
// modules/a/index.ts
```

```ts
import * from "./modules/a";
```

Reply:
we should handle this case too.

And since we are renaming a module, we should also rename its references in tsconfig.json like `"files"`/`"include"`/`"paths"` etc..

I think we should just call in `getEditsForFileRename` instead of computing something on the fly.